src/dataset-builder/dataset-builder-types.ts

```diff
@@ -8,7 +8,7 @@ import {
 let groupCount = 1;
 export const newCriteriaGroup = (): CriteriaGroup => {
   return {
-    name: `Group ${groupCount++}`,
+    id: groupCount++,
```

Comment:
Do we need to include `id`? What is that used for.

Reply:
In DomainCriteriaSelector, saveSelected uses the id!